drivers/mtd/gd55.c

```diff
@@ -0,0 +1,2065 @@
+                  startblock << GD55_SECTOR_SHIFT);
+        }
+      else
+        {
+          /* Erase each sector */
+
+          ret = gd55_erase_sector(priv, startblock);
+          if (ret < 0)
+            {
+              return ret;
+            }
+
+          startblock++;
+          blocksleft--;
+          finfo("Erased 4kbytes at address 0x%08" PRIx32 "\n",
+                  startblock << GD55_SECTOR_SHIFT);
+        }
+    }
+#endif
+
+  gd55_unlock(priv);
+
+  return (int)nblocks;
+}
+
+/****************************************************************************
+ * Name: gd55_bread
+ *
+ * Description:
+ *   Read a number of blocks of data.
+ *
+ * Input Parameters:
+ *   dev        - a reference to the device structure
+ *   startblock - start block of the memory to read
+ *   nblocks    - nblocks to read
+ *   buf        - pointer to the buffer to store the read data
+ *
+ * Returned Value:
+ *   Size of the data read
+ *
+ ****************************************************************************/
+
+ssize_t gd55_bread(FAR struct mtd_dev_s *dev, off_t startblock,
+                   size_t nblocks, FAR uint8_t *buf)
+{
+  ssize_t nbytes;
+
+  finfo("startblock: %08lx nblocks: %d\n", (long)startblock, (int)nblocks);
+
+  /* On this device, we can handle the block read just like the
+   * byte-oriented read
+   */
+
+#ifdef CONFIG_MTD_GD55_SECTOR512
+  nbytes = gd55_read(dev, startblock << GD55_SECTOR512_SHIFT,
+                     nblocks << GD55_SECTOR512_SHIFT, buf);
+  if (nbytes > 0)
+    {
+      nbytes >>= GD55_SECTOR512_SHIFT;
+    }
+#else
+  nbytes = gd55_read(dev, startblock << GD55_PAGE_SHIFT,
+                       nblocks << GD55_PAGE_SHIFT, buf);
+  if (nbytes > 0)
+    {
+      nbytes >>= GD55_PAGE_SHIFT;
+    }
+#endif
+
+  return nbytes;
+}
+
+/****************************************************************************
+ * Name: gd55_bwrite
+ *
+ * Description:
+ *   Write a number of blocks of data.
+ *
+ * Input Parameters:
+ *   dev        - a reference to the device structure
+ *   startblock - start block of the memory to write
+ *   nblocks    - nblocks to write
+ *   buf        - pointer to the buffer with the data to write
+ *
+ * Returned Value:
+ *   Size of the data written
+ *
+ ****************************************************************************/
+
+ssize_t gd55_bwrite(FAR struct mtd_dev_s *dev, off_t startblock,
+                    size_t nblocks, FAR const uint8_t *buf)
+{
+  FAR struct gd55_dev_s *priv = (FAR struct gd55_dev_s *)dev;
+  int                   ret;
+
+  finfo("startblock: %08lx nblocks: %d\n", (long)startblock, (int)nblocks);
+
+  /* Lock the QuadSPI bus and write all of the pages to FLASH */
+
+  gd55_lock(priv);
+
+#if defined(CONFIG_MTD_GD55_SECTOR512)
+  ret = gd55_write_cache(priv, buf, startblock, nblocks);
+  if (ret < 0)
+    {
+      ferr("ERROR: gd55_write_cache failed: %d\n", ret);
+    }
+
```

Comment:
remove blank line.

Reply:
Not quite? If CONFIG_MTD_GD55_SECTOR512 is set, style will need a blank line after the if {} code, so perhaps it should be that the #else moves?

Reply:
done